roles/ceph-monitor/tasks/main.yml

```diff
@@ -0,0 +1,102 @@
+---
+# we're storing the monitor_secret in a file on the first monitor node
+- name: check if mon_secrets file exists
+  stat: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  failed_when: false
+  register: mon_file_check
+
+- name: create monitor initial keyring
+  command: ceph-authtool /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }} --create-keyring --name=mon. --gen-key --cap mon 'allow *'
+           creates=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to:  "{{ groups['mons'][0] }}"
+  when: mon_file_check.stat.exists == false
+
+- name: fetch contents of mon_secret file
+  slurp: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  register: mon_secret_file
+
+- name: copy mon_secret to all hosts
+  copy:
+    dest: /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+    content: "{{ hostvars[groups['mons'][0]]['mon_secret_file']['content'] | b64decode }}"
+
+- name: set initial monitor key permissions
+  file: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+        mode=0600
+        owner=root
+        group=root
+
+- name: create monitor directory
+  file: path=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}
+        state=directory
+        owner=root
+        group=root
+        mode=0755
+
+- name: ceph monitor mkfs
+  command: ceph-mon --mkfs -i {{ ansible_hostname }} --fsid {{ fsid_file.content | b64decode }} --keyring /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+           creates=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}/keyring
+
+- name: activate monitor with upstart
+  file: path=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}/{{ item }}
+        state=touch
+        owner=root
+        group=root
+        mode=0600
+  with_items:
+    - done
+    - upstart
+  changed_when: false
+
+- name: ensure ceph-mon service enabled and started
+  service: name=ceph-mon
+           state=started
+           enabled=yes
+           args=id={{ ansible_hostname }}
+
+# NOTE (leseb): wait for mon discovery and quorum resolution
+# the admin key is not instantanely created so we have to wait a bit
+- name: wait for client.admin key exists
+  wait_for: path=/etc/ceph/ceph.client.admin.keyring
+
+- name: create openstack pool
+  command: rados mkpool {{ item }}
+  with_items:
+    - "{{ ceph_monitor.openstack_block_storage_pool }}"
+  failed_when: false
+
+- name: create openstack keys
+  command: ceph auth get-or-create {{ item.name }} {{ item.value }} -o /etc/ceph/ceph.{{ item.name }}.keyring
+           creates=/etc/ceph/ceph.{{ item.name }}.keyring
+  with_items: ceph_monitor.openstack_keys
+
+- name: find ceph keys
+  shell: ls -1 /etc/ceph/*.keyring
```

Comment:
hm.. if you are referencing this line: https://github.com/blueboxgroup/ursula/pull/1098/files#diff-fbf6d60f8ff9940f0621a8a5c45c392eR35

```
name: check for a ceph socket
stat: path=/var/run/ceph/*.asok
changed_when: false
failed_when: false
register: socket
```

I just tested it:
```
TASK: [ceph-common | check for a ceph socket] *********************************
ok: [xxx.x.x.xx] => {"changed": false, "failed": false, "failed_when_result": false, "stat": {"exists": false}}
```

but there was an .asok file in there..
```
c1:/var/run/ceph# ll
total 0
drwxr-xr-x  3 root root  80 Aug  5 11:43 ./
drwxr-xr-x 15 root root 620 Aug  6 23:15 ../
srwxr-xr-x  1 root root   0 Aug  5 11:43 ceph-mon.qint-lon02-c1.asok=
drw-r--r--  2 root root  40 Aug  6 23:15 rbd-clients/
```

Will have to relook at this..